Mage.Sets/src/mage/cards/s/SilverFurMaster.java

```diff
@@ -0,0 +1,95 @@
+package mage.cards.s;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.common.continuous.BoostControlledEffect;
+import mage.abilities.effects.common.cost.CostModificationEffectImpl;
+import mage.abilities.keyword.NinjutsuAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.CostModificationType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.Zone;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.Predicates;
+import mage.game.Game;
+import mage.util.CardUtil;
+
+import java.util.UUID;
+
+/**
+ * @author Addictiveme
+ */
+public final class SilverFurMaster extends CardImpl {
+
+		private static final FilterCreaturePermanent filter = new FilterCreaturePermanent("Ninja and Rogue creatures");
+
+		static {
+			filter.add(Predicates.or(
+			        SubType.NINJA.getPredicate(),
+			        SubType.ROGUE.getPredicate()
+			));
+		}
+
+		public SilverFurMaster(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{U}{B}");
+
+        this.subtype.add(SubType.RAT);
+        this.subtype.add(SubType.NINJA);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Ninjutsu {U}{B}
+        this.addAbility(new NinjutsuAbility("{U}{B}"));
+
+        // Ninjutsu abilities you activate cost {1} less to activate.
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new SilverFurMasterCostModificationEffect()));
```

Comment:
You can use `AbilitiesCostReductionControllerEffect` here instead of making a new class.

Reply:
Oh nice, that's much easier. I've changed the code to use that instead.